venafi/provider.go

```diff
@@ -272,11 +280,44 @@ func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}
 		cfg.Credentials.AccessToken = accessToken
 
 	} else if clientCertMethod {
+		if p12CertData != "" && p12Certificate != "" {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  messageVenafiClientInitFailed,
+				Detail:   fmt.Sprintf("%s: Only one of p12_cert_data and p12_cert_filename should be provided", messageVenafiConfigFailed),
+			})
+			return nil, diags
+		}
+		var cert []byte
+		if p12CertData != "" {
+			data, err := base64.StdEncoding.DecodeString(p12CertData)
+			if err != nil {
+				diags = append(diags, diag.Diagnostic{
+					Severity: diag.Error,
+					Summary:  messageVenafiClientInitFailed,
+					Detail:   fmt.Sprintf("%s: Failed to decode p12 Certificate Data", messageVenafiConfigFailed),
+				})
+				return nil, diags
+			}
+			cert = data
+		}
+		if p12Certificate != "" {
+			data, err := os.ReadFile(p12Certificate)
+			if err != nil {
+				diags = append(diags, diag.Diagnostic{
+					Severity: diag.Error,
+					Summary:  messageVenafiClientInitFailed,
+					Detail:   fmt.Sprintf("%s: Failed to read p12 Certificate file", messageVenafiConfigFailed),
```

Comment:
We must keep the original error message behavior as it's in the removed line 397 of this file, maybe similar to the next proposal:
`Detail:   fmt.Sprintf("%s: %s", messageVenafiConfigFailed, fmt.Errorf("unable to read PKCS#12 file at [%s]: %w", p12Certificate, err)),`
Similar consideration should be taken for the added line 298